meta-client/src/main/java/org/apache/helix/metaclient/recipes/leaderelection/LeaderElectionListenerInterface.java

```diff
@@ -0,0 +1,32 @@
+package org.apache.helix.metaclient.recipes.leaderelection;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * It provides APIs for listener listening on events like a new leader is created or current
+ * leader node is deleted.
+ */
+public interface LeaderElectionListenerInterface {
+  // When new leader is elected:
+  //                             noLeader (null)                  ->    has leader (new leader name)
+  // When existing leader not leader anymore:
+  //                             has Leader (prevleader name)     ->    no leader (null)
+  public void onLeadershipChange(String leaderPath, String prevLeader, String curLeader);
```

Comment:
Going by the comment since either curLeader/prevLeader will be null on acquire/relinquish, I feel it's better to modify the signature of the function to use `onLeadershipChange(String leaderPath, String leader, Enum EVENT( RELINQUISH/ACQUIRE) event`. 

In case I misunderstood, can you rename `noLeader/ hasLeader` to match the context of `prevLeader/cureLeader`

Reply:
As discussed offline. This API is more aligned with current use case.